packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -349,3 +349,117 @@ class AOTSnapshotter {
     return fs.path.dirname(fs.path.fromUri(packageMap.map[package]));
   }
 }
+
+class CoreJITSnapshotter {
+  /// Builds a "Core JIT" VM snapshot of the specified kernel. This snapshot
+  /// includes data as well as either machine code or DBC, depending on build
+  /// configuration.
+  Future<int> build({
+    @required TargetPlatform platform,
+    @required BuildMode buildMode,
+    @required String mainPath,
+    @required String packagesPath,
+    @required String outputPath,
+    List<String> extraGenSnapshotOptions = const <String>[],
+  }) async {
+    if (!_isValidCoreJitPlatform(platform)) {
+      printError('${getNameForTargetPlatform(platform)} does not support Core JIT compilation.');
+      return 1;
+    }
+
+    final Directory outputDir = fs.directory(outputPath);
+    outputDir.createSync(recursive: true);
+
+    final List<String> inputPaths = <String>[mainPath];
+    final Set<String> outputPaths = new Set<String>();
+
+    final String depfilePath = fs.path.join(outputDir.path, 'snapshot.d');
+    final List<String> genSnapshotArgs = <String>[
+      '--reify-generic-functions',
+      '--strong',
+    ];
+    if (extraGenSnapshotOptions != null && extraGenSnapshotOptions.isNotEmpty) {
+      printTrace('Extra gen_snapshot options: $extraGenSnapshotOptions');
+      genSnapshotArgs.addAll(extraGenSnapshotOptions);
+    }
+
+    // Blob Core JIT snapshot.
+    final String vmSnapshotData = fs.path.join(outputDir.path, 'vm_snapshot_data');
+    final String isolateSnapshotData = fs.path.join(outputDir.path, 'isolate_snapshot_data');
+    final String vmSnapshotInstructions = fs.path.join(outputDir.path, 'vm_snapshot_instr');
+    final String isolateSnapshotInstructions = fs.path.join(outputDir.path, 'isolate_snapshot_instr');
+    outputPaths.addAll(<String>[vmSnapshotData, isolateSnapshotData, vmSnapshotInstructions, isolateSnapshotInstructions]);
+    genSnapshotArgs.addAll(<String>[
+      '--snapshot_kind=core-jit',
+      '--vm_snapshot_data=$vmSnapshotData',
+      '--isolate_snapshot_data=$isolateSnapshotData',
+      '--vm_snapshot_instructions=$vmSnapshotInstructions',
+      '--isolate_snapshot_instructions=$isolateSnapshotInstructions',
+      '--load_compilation_trace=trace.txt',
+    ]);
+
+    if (platform == TargetPlatform.android_arm) {
+      // Use softfp for Android armv7 devices.
+      // Note that this is the default for armv7 iOS builds, but harmless to set.
```

Comment:
Kill this line of the comment since this is Android-specific; the other two are good as-is.